sherpa/image/ds9_backend.py

```diff
@@ -23,9 +23,16 @@
 from os import access, R_OK
 from sherpa.utils.err import DS9Err
 
+import six
+
 imager = DS9.DS9Win(DS9._DefTemplate, False)
 
 
+# TODO: the except blocks would ideally catch explicit errors; the present
+#       catch-anything approach means that we lose potentially-useful
+#       information on the type of error.
```

Comment:
Yeah, unfortunately Python2 lacks a standard mechanism for chaining exceptions, but I believe they fixed that, finally, in Python 3. Not sure whether it makes sense to fix it now in a 2+3 fashion.